src/main/java/io/vavr/collection/Traversable.java

```diff
@@ -1704,6 +1704,45 @@ default Number sum() {
      */
     <U> Traversable<U> zipWithIndex(BiFunction<? super T, ? super Integer, ? extends U> mapper);
 
+    /**
+     * Converts this collection to a {@link Map}.
+     *
+     * @param f A function that maps an element to a key/value pair represented by Tuple2
+     * @param merge A function that merges values that are associated with the same key
+     * @param <K>         The key type
+     * @param <V>         The value type
+     * @return A new {@link HashMap}.
+     */
+    default <K, V, E extends Function<? super T, ? extends Tuple2<? extends K, ? extends V>>> Map<K, V> toMap(
+            E f,
```

Comment:
This helps to disambiguate between this method and [`Value.toMap(Function, Function)`](https://www.javadoc.io/doc/io.vavr/vavr/0.9.2/io/vavr/Value.html#toMap-java.util.function.Function-java.util.function.Function-).

```
src/main/java/io/vavr/collection/Traversable.java:1716: warning: [overloads] <K#1,V#1>toMap(Function<? super T#1,? extends Tuple2<? extends K#1,? extends V#1>>,BiFunction<? super V#1,? super V#1,? extends V#1>) in Traversable is potentially ambiguous with <K#2,V#2>toMap(Function<? super T#2,? extends K#2>,Function<? super T#2,? extends V#2>) in Value
    default <K, V> Map<K, V> toMap(
                             ^
  where K#1,V#1,T#1,K#2,V#2,T#2 are type-variables:
    K#1 extends Object declared in method <K#1,V#1>toMap(Function<? super T#1,? extends Tuple2<? extends K#1,? extends V#1>>,BiFunction<? super V#1,? super V#1,? extends V#1>)
    V#1 extends Object declared in method <K#1,V#1>toMap(Function<? super T#1,? extends Tuple2<? extends K#1,? extends V#1>>,BiFunction<? super V#1,? super V#1,? extends V#1>)
    T#1 extends Object declared in interface Traversable
    K#2 extends Object declared in method <K#2,V#2>toMap(Function<? super T#2,? extends K#2>,Function<? super T#2,? extends V#2>)
    V#2 extends Object declared in method <K#2,V#2>toMap(Function<? super T#2,? extends K#2>,Function<? super T#2,? extends V#2>)
    T#2 extends Object declared in interface Value
error: warnings found and -Werror specified
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
1 error
1 warning
```
The only alternatives I could find are:
1) change the method name to something like `toMapWithMerge` (but kind of defeats the purpose of aligning this API with Java's);
2) use `SuppressWarnings("overloads")` instead.

Reply:
My experience is that similar method signatures will cause problems when used in conjunction with lambdas. Please delete this method for now. We will keep only the second method (below).